pytket/tests/predicates_test.py

```diff
@@ -1092,6 +1092,26 @@ def test_greedy_pauli_synth() -> None:
     assert GreedyPauliSimp().apply(c)
 
 
+def test_get_pre_conditions() -> None:
+    pre_cons = GreedyPauliSimp().get_preconditions()
+    gate_set = pre_cons[0].gate_set  # type: ignore
+    assert OpType.CX in gate_set
+    assert OpType.Measure in gate_set
+
+
+def test_get_post_conditions() -> None:
+    gate_set = {OpType.CX, OpType.Rz, OpType.H, OpType.Reset, OpType.Measure}
+    post_cons = AutoRebase(gate_set).get_postconditions()
+    assert post_cons[0].gate_set == gate_set  # type: ignore
+
+
+def test_get_gate_set() -> None:
+    gate_set = GreedyPauliSimp().get_gate_set()
+    assert OpType.CX in gate_set  # type: ignore
+    assert OpType.Measure in gate_set  # type: ignore
```

Comment:
Should be able to avoid the `type: ignore`s by first asserting that `gate_set is not None`.